### PR TITLE
bintray: url encode where needed

### PR DESCRIPTION
**Pull request by Bo98**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

There is no guarantee that what we are passing here is URL safe. Therefore, we should URL encode the input.

### CODE REVIEW

**Comment by Bo98** [comment]
I have not touched `BottlePublisher`. I do not think it is used anywhere anymore.

File: Library/Homebrew/utils/bottles.rb [hunk 1]
```diff
@@ -75,7 +75,7 @@ def self.package(formula_name)
         package_name = formula_name.to_s.dup
         package_name.tr!("+", "x")
         package_name.sub!(/(.)@(\d)/, "\\1:\\2") # Handle foo@1.2 style formulae.
-        package_name
+        ERB::Util.url_encode(package_name)
```
Comment (Bo98) [inline Library/Homebrew/utils/bottles.rb:78]: I suspect this actually fixes some of the hacks above like `+` (which would have been interpreted as a space if not URL encoded). Too late to remove them now though.

**Comment by jonchang** [comment]
> I have not touched `BottlePublisher`. I do not think it is used anywhere anymore.

I think this entire class can go away.

**Comment by jonchang** [comment]
Is there a difference between `URI.encode_www_form_component` vs `ERB::Util.url_encode`? Otherwise LGTM

**Comment by Bo98** [comment]
General rule is that `encode_www_form*` is for form data - that is query parameters (things after the `?`) or POST data.

`ERB::Util.url_encode` is for the rest of the URL.

They follow different standards for encoding. The notable difference (albeit not one we will face here) is that form encoding turns space characters to `+` and URL encoding turns space characters to `%20`.

**Comment by Bo98** [comment]
There's perhaps an argument for URL encoding in `Utils::Bottles::Bintray.package` too (I suspect some of the swapping it already does is due to the lack of encoding) but I kept this change small and is _probably_ something we'll not need in Homebrew/core - though I suppose I would've said the same about versions a week ago.

**Pull request merged**